include/FilesystemManagement.h

```diff
@@ -0,0 +1,156 @@
+    /// @param OldFilePath The existing path to the file (including the filename) to be copied.
+    /// @param NewFilePath The path (including the filename) to where the file should be copied.
+    /// @param FailIfExists If true the operation will fail if a file with the target name already exists.
+    /// @return Returns true if the file was successfully copied, false otherwise.
+    [[nodiscard]]
+    ModifyResult MEZZ_LIB CopyFile(const StringView OldFilePath, const StringView NewFilePath,
+                                   const Boole FailIfExists);
+    /// @brief Moves a file on disk from one location to another.
+    /// @remarks This function can be used to rename files.
+    /// @param OldFilePath The existing path to the file (including the filename) to be moved.
+    /// @param NewFilePath The path (including the filename) to where the file should be named.
+    /// @param FailIfExists If true the operation will fail if a file with the target name already exists.
+    /// @return Returns true if the file was successfully moved, false otherwise.
+    [[nodiscard]]
+    ModifyResult MEZZ_LIB MoveFile(const StringView OldFilePath, const StringView NewFilePath,
+                                   const Boole FailIfExists);
+    /// @brief Deletes a file existing on the filesystem.
+    /// @param FilePath The existing path to the file (including the filename) to be deleted.
+    /// @return Returns true if the operation was successful, false if it failed.
+    [[nodiscard]]
+    ModifyResult MEZZ_LIB RemoveFile(const StringView FilePath);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Symlinks
+
+    /// @brief Creates a symbolic link to a file on disk.
+    /// @param SymPath A path (including the name of the symbolic link) to where the link should be placed.
+    /// @param TargetPath A path to where the symbolic link will point to.
+    /// @return Returns true if the operation was successful, false otherwise.
+    [[nodiscard]]
+    ModifyResult MEZZ_LIB CreateSymlink(const StringView SymPath, const StringView TargetPath);
+    /// @brief Creates a symbolic link to a directory on disk.
+    /// @param SymPath A path (including the name of the symbolic link) to where the link should be placed.
+    /// @param TargetPath A path to where the symbolic link will point to.
+    /// @return Returns true if the operation was successful, false otherwise.
+    [[nodiscard]]
+    ModifyResult MEZZ_LIB CreateDirectorySymlink(const StringView SymPath, const StringView TargetPath);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Basic Directory Management
+
+    /// @brief Verifies the existence of a folder.
+    /// @param DirectoryPath The path and name of the folder to check for.
+    /// @return Returns true if the folder at the specified location exists, false otherwise.
```

Comment:
What is returned if it is a file?
I  presume false it returned, but that should be documented.

Reply:
I changed location -> directory.  I think it was clear enough before, but this way we're using common vocabulary at least.